include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -45,32 +64,54 @@ namespace sparrow
         explicit arrow_proxy(ArrowArray&& array, ArrowSchema&& schema);
         /// Constructs an arrow_proxy which takes the ownership of the ArrowArray and uses the provided
         /// ArrowSchema. The array is released when the arrow_proxy is destroyed. The schema is not released.
-        explicit arrow_proxy(ArrowArray&& array, ArrowSchema* schema);
+        explicit arrow_proxy(ArrowArray&& array, ArrowSchema* schema, ownership schema_ownership);
```

Comment:
I dont like that the ownership has to be specified only for one of the parameters, I would suggest to either request it always, with a default value to shorten code, or to only provide one constructor with pointers and the ownership options.

Also note that [`arrow_data_ownership`](https://github.com/man-group/sparrow/blob/main/include/sparrow/c_interface.hpp#L93) lets the user specify both ownerships already, you should never need to use `ownership` alone in these interfaces.

Reply:
In that case, the proxy always takes the ownership and also hold the data in the variant. The user should not give a misleading information.

Reply:
I don't understand why we need an additional member for specifying the ownership in the API:
- either the user "moves" the data / schema into the proxy, and therefore transfers the ownership
- or it passes a pointer (because the same structure may be used in different arrow_proxy objects) and therefore keeps the ownership. The structure must be released by the user calling `release` on the "initial" C structure.

I don't see a usecase where the user would like to transfer the owership *and* keep a pointer on the original C structure.

Reply:
I finally removed the ownership parameter.
According to the Apache Arrow documentation:
https://arrow.apache.org/docs/dev/format/CDataInterface.html#member-allocation

> It is intended for the base structure to be stack- or heap-allocated by the consumer. In this case, the producer API should take a pointer to the consumer-allocated structure.

So, as a consumer, we never get a pointer from a producer third party.